v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs

```diff
@@ -0,0 +1,167 @@
+    {
+      // Arrange
+      var numberOfSnapshotModels = 4;
+
+      var snapshotModel = _fixture.CreateMany<TeamSnapshotViewModel>(numberOfSnapshotModels);
+      var snapshotReturnedFromService = _fixture.Create<Dictionary<string, List<EmployeeSnapshotDto>>>();
+
+      _dashboardService.GetTeamSnapshotDashboardEvents().Returns(snapshotReturnedFromService);
+
+      _mapper.Map<Dictionary<string, List<EmployeeSnapshotDto>>, List<TeamSnapshotViewModel>>(snapshotReturnedFromService).Returns(snapshotModel);
+
+      // Act
+      var result = _dashboardController.GetDashboardSnapshot();
+
+      // Assert
+      RetrieveValueFromResult<List<TeamSnapshotViewModel>>(result);
+
+      _dashboardService.Received(1).GetTeamSnapshotDashboardEvents();
+
+      _mapper.Received(1).Map<Dictionary<string, List<EmployeeSnapshotDto>>, TeamSnapshotViewModel>(Arg.Is(snapshotReturnedFromService));      
+    }
+
+    [Fact]
+    public void GetEmployeeEvents_WhenCalled_ReturnsEmployeeEvents()
+    {
+      // Arrange
+      var employeeId = 88;
+      var numberOfEvents = 5;
+      var searchDate = DateTime.Now;
+
+      var eventsReturnedFromService = _fixture.CreateMany<EventDto>(numberOfEvents).ToList();
+      var eventModels = _fixture.CreateMany<DashboardEventViewModel>(numberOfEvents).ToList();
+
+      _dashboardService.GetEmployeeDashboardEvents(employeeId, searchDate).Returns(eventsReturnedFromService);
+
+      _mapper.Map<IList<EventDto>, List<DashboardEventViewModel>>(Arg.Is(eventsReturnedFromService)).Returns(eventModels);
+
+      _employeeCredentials.GetUserId().Returns(employeeId);
+
+      // Act
+      var result = _dashboardController.GetEmployeeEvents(searchDate);
+
+      // Assert
+      RetrieveValueFromResult<EmployeeEventViewModel>(result);
+
+      _dashboardService.Received(1).GetEmployeeDashboardEvents(Arg.Is<int>(x => x == employeeId), Arg.Is<DateTime>(x => x == searchDate));
+
+      _mapper.Received(1).Map<List<EventDto>, List<DashboardEventViewModel>>(Arg.Is(eventsReturnedFromService));      
+    }
+
+    [Fact]
+    public void GetEmployeeTeamSnapshot_WhenCalled_ReturnsEmployeeTeamSnapshot()
+    {
+      // Arrange
+      var employeeId = 41;
+      var numberOfSnapshotModels = 6;
+
+      var snapshotReturnedFromService = _fixture.Create<Dictionary<string, List<EmployeeSnapshotDto>>>();
+      var snapshotModels = _fixture.CreateMany<TeamSnapshotViewModel>(numberOfSnapshotModels);
+
+      _dashboardService.GetEmployeeSnapshotsByEmployeeId(Arg.Is(employeeId)).Returns(snapshotReturnedFromService);
+
+      _mapper.Map<Dictionary<string, List<EmployeeSnapshotDto>>, List<TeamSnapshotViewModel>>(snapshotReturnedFromService).Returns(snapshotModels);      
+
+      _employeeCredentials.GetUserId().Returns(employeeId);
+
+      // Act
+      var result = _dashboardController.GetEmployeeTeamSnapshot();
+
+      // Assert
+      RetrieveValueFromResult<List<EmployeeEventViewModel>>(result);
```

Comment:
I'll comment this on, but needs done throughout.

Reply:
Updated - Setting what's returned and when applicable checking the count is correct.